ebos/eclgenericoutputblackoilmodule.cc

```diff
@@ -944,50 +944,42 @@ setRestart(const data::Solution& sol,
-        cPolymer_[elemIdx] = sol.data("POLYMER")[globalDofIndex];
-    if (!cFoam_.empty() && sol.has("FOAM"))
-        cFoam_[elemIdx] = sol.data("FOAM")[globalDofIndex];
-    if (!cSalt_.empty() && sol.has("SALT"))
-        cSalt_[elemIdx] = sol.data("SALT")[globalDofIndex];
-    if (!pSalt_.empty() && sol.has("SALTP"))
-        pSalt_[elemIdx] = sol.data("SALTP")[globalDofIndex];
-    if (!permFact_.empty() && sol.has("PERMFACT"))
-        permFact_[elemIdx] = sol.data("PERMFACT")[globalDofIndex];
-    if (!soMax_.empty() && sol.has("SOMAX"))
-        soMax_[elemIdx] = sol.data("SOMAX")[globalDofIndex];
-    if (!pcSwMdcOw_.empty() && sol.has("PCSWM_OW"))
-        pcSwMdcOw_[elemIdx] = sol.data("PCSWM_OW")[globalDofIndex];
-    if (!krnSwMdcOw_.empty() && sol.has("KRNSW_OW"))
-        krnSwMdcOw_[elemIdx] = sol.data("KRNSW_OW")[globalDofIndex];
-    if (!pcSwMdcGo_.empty() && sol.has("PCSWM_GO"))
-        pcSwMdcGo_[elemIdx] = sol.data("PCSWM_GO")[globalDofIndex];
-    if (!krnSwMdcGo_.empty() && sol.has("KRNSW_GO"))
-        krnSwMdcGo_[elemIdx] = sol.data("KRNSW_GO")[globalDofIndex];
-    if (!ppcw_.empty() && sol.has("PPCW"))
-        ppcw_[elemIdx] = sol.data("PPCW")[globalDofIndex];
-    if (!cMicrobes_.empty() && sol.has("MICROBES"))
-        cMicrobes_[elemIdx] = sol.data("MICROBES")[globalDofIndex];
-    if (!cOxygen_.empty() && sol.has("OXYGEN"))
-        cOxygen_[elemIdx] = sol.data("OXYGEN")[globalDofIndex];
-    if (!cUrea_.empty() && sol.has("UREA"))
-        cUrea_[elemIdx] = sol.data("UREA")[globalDofIndex];
-    if (!cBiofilm_.empty() && sol.has("BIOFILM"))
-        cBiofilm_[elemIdx] = sol.data("BIOFILM")[globalDofIndex];
-    if (!cCalcite_.empty() && sol.has("CALCITE"))
-        cCalcite_[elemIdx] = sol.data("CALCITE")[globalDofIndex];
+    auto assign = [elemIdx, globalDofIndex, &sol](ScalarBuffer& data,
+                                                  const std::string& name)
+    {
+        if (!data.empty() && sol.has(name)) {
+            data[elemIdx] = sol.data(name)[globalDofIndex];
+        }
+    };
+
+    const std::vector<std::pair<std::string, ScalarBuffer*>> fields = {
+        {"BIOFILM", &cBiofilm_},
+        {"CALCITE",&cCalcite_},
+        {"FOAM", &cFoam_},
+        {"KRNSW_GO", &krnSwMdcGo_},
+        {"KRNSW_OW", &krnSwMdcOw_},
+        {"MICROBES", &cMicrobes_},
+        {"OXYGEN", &cOxygen_},
+        {"PCSWM_GO", &pcSwMdcGo_},
+        {"PCSWM_OW", &pcSwMdcOw_},
+        {"PERMFACT", &permFact_},
+        {"POLYMER", &cPolymer_},
+        {"PPCW", &ppcw_},
+        {"PRESSURE", &fluidPressure_},
+        {"RS", &rs_},
+        {"RSW", &rsw_},
+        {"RV", &rv_},
+        {"RVW", &rvw_},
+        {"SALT", &cSalt_},
+        {"SALTP", &pSalt_},
+        {"SOMAX", &soMax_},
+        {"TEMP", &temperature_},
+        {"UREA", &cUrea_},
+    };
+
+    std::for_each(fields.begin(), fields.end(),
+                  [assign](const std::pair<std::string, ScalarBuffer*>& p)
```

Comment:
I'd prefer to take the `assign` capture by reference here and use a generic lambda (`const auto& p`) instead of fully specifying the type of the `pair` parameter.

Reply:
done and done.